### PR TITLE
read_header

### PR DESCRIPTION
**Pull request by geosciz**

### CODE REVIEW

**Comment by rabernat** [comment]
Could the failing test be related to the pandas version?
https://travis-ci.org/rabernat/floater/jobs/242245642#L325

It's 0.20 on travis. What version of pandas do you have locally?

**Comment by geosciz** [comment]
@rabernat 0.19.2

**Comment by geosciz** [comment]
@rabernat I agree it's `pandas` version issue. See this line: https://travis-ci.org/rabernat/floater/jobs/242245642#L563. The problem does not originate from the code inside floater, but from something internally in `core` directory of `pandas`. In the test, I create an index ranging from 1 to 81, however, `indexer` variable here (created by `pandas` internally): https://travis-ci.org/rabernat/floater/jobs/242245642#L574 ranges from 1 to 80.

**Comment by rabernat** [comment]
@geosciz I think you might need to rebase your branch.

File: floater/test/test_generators.py [hunk 1]
```diff
@@ -189,14 +189,14 @@ def test_pickling_with_land(fs_with_land, tmpdir):
...
-    lat = np.linspace(-4, 4, 9, dtype=np.float32)
-    land_mask = np.zeros(81, dtype=bool)==False
+    lon = np.arange(0, 9, dtype=np.float32)
+    lat = np.arange(-4, 5, dtype=np.float32)
+    land_mask = np.full(81, True, dtype=bool)
     land_mask.shape = (len(lat), len(lon))
     land_mask[:,0:2] = False
     model_grid = {'lon': lon, 'lat': lat, 'land_mask': land_mask}
-    fs_none = gen.FloatSet(xlim=(0, 9), ylim=(-4, 5), dx=1.0, dy=1.0)
-    fs_mask = gen.FloatSet(xlim=(0, 9), ylim=(-4, 5), dx=1.0, dy=1.0, model_grid=model_grid)
```
Comment (rabernat) [inline floater/test/test_generators.py]: Can you explain why you made these changes?

**Comment by rabernat** [comment]
Does this pass your local tests as well?

**Comment by geosciz** [comment]
@rabernat I updated `pandas` and it passed my local test.

File: floater/test/test_generators.py (hunk 1 above)
Comment (geosciz) [inline floater/test/test_generators.py]: no specific purpose, some simplification I made when I checked my test code yesterday.

**Comment by rabernat** [comment]
Look at these tests:
https://travis-ci.org/rabernat/floater

**Comment by rabernat** [comment]
If you rebase again, you will get my new test maxtrix

File: floater/utils.py [hunk 2]
```diff
@@ -252,12 +252,12 @@ def floats_to_netcdf(input_dir, output_fname,
...
     float_files = glob(os.path.join(input_dir, match_pattern))
     float_timesteps = sorted(list({int(float_file[-22:-12]) for float_file in float_files}))
+    float_columns = ['npart', 'time', 'x', 'y', 'z', 'u', 'v', 'vort']
 
     for float_timestep in tqdm(float_timesteps):
         input_path = os.path.join(input_dir, '%s.%010d.*.csv' % (float_file_prefix, float_timestep))
         df = dd.read_csv(input_path)
         if df.columns.values[0] is not 'npart':
-            float_columns = ['npart', 'time', 'x', 'y', 'z', 'u', 'v', 'vort']
             df = dd.read_csv(input_path, names=float_columns, header=None)
```
Comment (rabernat) [inline floater/utils.py]: I do not like the fact that we potential read the data twice! That could really slow things down.

Could we not just open the first file directly and peek at the header?

Comment (geosciz) [inline floater/utils.py]: I included a line to get the `type` of header to fix this issue.

File: floater/utils.py [hunk 3]
```diff
@@ -251,13 +251,13 @@ def floats_to_netcdf(input_dir, output_fname,
 
     match_pattern = float_file_prefix + '.*.csv'
     float_files = glob(os.path.join(input_dir, match_pattern))
-    header_type = type(dd.read_csv(float_files[0]).columns[0])
+    float_header = dd.read_csv(float_files[0]).columns
```
Comment (rabernat) [inline floater/utils.py]: I recommend you replace this with a pandas call which only reads two rows, i.e.
```python
pd.read_csv(float_files[0], nrows=2)
```
https://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html

**Comment by rabernat** [comment]
I know you have been working on this a long time and are eager to finish it off.

But why don't we just add the test data with headers to this PR? It really belongs here.

**Comment by geosciz** [comment]
@rabernat No problem, I am going to add the test data today.

File: floater/utils.py [hunk 4]
```diff
@@ -251,7 +252,7 @@ def floats_to_netcdf(input_dir, output_fname,
 
     match_pattern = float_file_prefix + '.*.csv'
     float_files = glob(os.path.join(input_dir, match_pattern))
-    float_header = dd.read_csv(float_files[0]).columns
+    float_header = pd.read_csv(float_files[0], nrows=0).columns
```
Comment (rabernat) [inline floater/utils.py]: Does this really work with `nrows=0`? I'm surprised...

Comment (geosciz) [inline floater/utils.py]: Yes. I tested it using different values 2, 1, and 0, and all of them work. So I picked 0 here.

Comment (rabernat) [inline floater/utils.py]: ok, well I guess we are ready to go!

**Pull request merged**